File: app/src/main/java/org/schulcloud/mobile/data/model/Directory.java
package org.schulcloud.mobile.data.model;

import org.parceler.Transient;

import io.realm.RealmModel;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.RealmClass;

@RealmClass
public class Directory implements RealmModel {
    @PrimaryKey
    public String name;
    public String path;
}

File: app/src/main/java/org/schulcloud/mobile/models/file/Directory.kt
package org.schulcloud.mobile.models.file

import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import org.schulcloud.mobile.models.base.HasId

open class Directory : RealmObject(), HasId {
    override val id: String
        get() = key ?: ""

    @PrimaryKey
    var key: String? = null
    var name: String? = null
    var path: String? = null
}
